lib/active_fedora/associations/directly_contains_association.rb

```diff
@@ -0,0 +1,54 @@
+module ActiveFedora
+  module Associations
+    class DirectlyContainsAssociation < CollectionAssociation #:nodoc:
+
+      def insert_record(record, force = true, validate = true)
+        container.save!
+        if record.new_record?
+          if force
+            record.save!
+          else
+            return false unless record.save(validate: validate)
+          end
+        end
+
+        return true
+      end
+
+      def reader
+        @records ||= ContainerProxy.new(self)
+      end
+
+      def find_target
+        uris = owner.resource.query(predicate: options[:predicate]).map { |r| r.object.to_s }
+        if klass <= ActiveFedora::File # a subclass of file
+          uris.map { |uri| klass.new(uri) }
```

Comment:
I think there's a lot of benefit in just sending messages and not having to do a class check to get our functionality. Happy to be overridden.